src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonIEnumerableDefaultConverter.cs

```diff
@@ -36,7 +37,7 @@ protected static JsonConverter<TElement> GetElementConverter(ref WriteStack stat
             Type typeToConvert,
             JsonSerializerOptions options,
             ref ReadStack state,
-            out TCollection value)
+            [MaybeNullWhen(false)] out TCollection value)
```

Comment:
With this, once https://github.com/dotnet/runtime/pull/32090 is merged, we can remove the `!` in classes like:
JsonIEnumerableDefaultConverter.cs
JsonDictionaryDefaultConverter.cs
etc.

Everywhere we do:
`value = default!;`